rasa/shared/core/training_data/story_reader/yaml_story_reader.py

```diff
@@ -165,21 +166,28 @@ def is_stories_file(cls, file_path: Union[Text, Path]) -> bool:
 
     @classmethod
     def is_key_in_yaml(cls, file_path: Union[Text, Path], *keys: Text) -> bool:
-        """Check if all keys are contained in the parsed dictionary from a yaml file.
+        """Check if any of the keys is contained in the root object of the yaml file.
 
         Arguments:
             file_path: path to the yaml file
             keys: keys to look for
 
         Returns:
-              `True` if all the keys are contained in the file, `False` otherwise.
+              `True` if at least one of the keys is found, `False` otherwise.
 
         Raises:
-            YamlException: if the file seems to be a YAML file (extension) but
-                can not be read / parsed.
+            FileNotFoundException: if the file cannot be found.
         """
-        content = rasa.shared.utils.io.read_yaml_file(file_path)
-        return any(key in content for key in keys)
+        try:
+            with open(file_path) as file:
+                return any(
+                    any(line.lstrip().startswith(f"{key}:") for key in keys) for line in file
```

Comment:
I've also added `.lstrip()` here because apparently this is a valid YAML:
```
        stories:
        - story: simple_story_without_checkpoint
          steps:
          - intent: simple
          - action: utter_default
          - action: utter_greet
```